### PR TITLE
add float_pow

### PR DESCRIPTION
**Pull request by pwang200**
Add a new float_pow function for calculation of powers

Add tests including edge cases

### CODE REVIEW

File: wasm-host/src/host_functions_wamr.rs [hunk 1]
```diff
@@ -570,6 +570,39 @@ pub fn float_divide(
     pack_out_float(r, env, out_buf)
 }
 
+pub fn float_pow(
+    env: wasm_exec_env_t,
+    in_buf: *const u8,
+    in_int: i32,
+    out_buf: *mut u8,
+    rounting_mode: i32,
```
[the remark this answers is not in the document]
Comment (pwang200) [reply wasm-host/src/host_functions_wamr.rs:578]: Yes. They are placeholders for now.

**Comment by pwang200** [comment]
@intelliot as discussed in the standup, I added buffer length to all float host functions

File: wasm-host/src/host_functions_wamr.rs [hunk 2]
```diff
@@ -435,7 +435,8 @@ pub fn float_from_int(
     env: wasm_exec_env_t,
     in_int: i64,
     out_buf: *mut u8,
```
[the remark this answers is not in the document]
Comment (pwang200) [reply wasm-host/src/host_functions_wamr.rs:437]: agree. I plan to have a separate PR for that.
